remme/tp/pub_key.py

```diff
@@ -0,0 +1,124 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+
+import datetime
+import logging
+import binascii
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives.serialization import load_pem_public_key
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+
+from remme.tp.basic import BasicHandler, get_data
+from remme.tp.account import AccountHandler, get_account_by_address
+
+from remme.protos.pub_key_pb2 import (
+    PubKeyStorage,
+    NewPubKeyPayload, RevokePubKeyPayload, PubKeyMethod
+)
+from remme.shared.singleton import singleton
+
+LOGGER = logging.getLogger(__name__)
+
+FAMILY_NAME = 'pub_key'
+FAMILY_VERSIONS = ['0.1']
+
+CERT_ORGANIZATION = 'REMME'
+CERT_MAX_VALIDITY_DAYS = 365
+CERT_MAX_VALIDITY = datetime.timedelta(CERT_MAX_VALIDITY_DAYS)
+
+CERT_STORE_PRICE = 10
+
+
+@singleton
+class PubKeyHandler(BasicHandler):
+    def __init__(self):
+        super().__init__(FAMILY_NAME, FAMILY_VERSIONS)
+
+    def get_state_processor(self):
+        return {
+            PubKeyMethod.STORE: {
+                'pb_class': NewPubKeyPayload,
+                'processor': self._store_pub_key
+            },
+            PubKeyMethod.REVOKE: {
+                'pb_class': RevokePubKeyPayload,
+                'processor': self._revoke_pub_key
+            }
+        }
+
+    def _store_pub_key(self, context, signer_pubkey, transaction_payload):
+        address = self.make_address_from_data(transaction_payload.public_key)
+        LOGGER.info('Cert address {}'.format(address))
+        data = get_data(context, PubKeyStorage, address)
+        if data:
+            raise InvalidTransaction('This pub key is already registered.')
+
+        cert_signer_pubkey = load_pem_public_key(transaction_payload.public_key.encode('utf-8'),
+                                                 backend=default_backend())
+        try:
+            ehs_bytes = binascii.unhexlify(transaction_payload.entity_hash_signature)
+            eh_bytes = binascii.unhexlify(transaction_payload.entity_hash)
+        except binascii.Error:
+            LOGGER.debug(f'entity_hash_signature {transaction_payload.entity_hash_signature}')
+            LOGGER.debug(f'entity_hash {transaction_payload.entity_hash}')
+            raise InvalidTransaction('Entity hash or signature not a hex format')
+
+        # NOTE: For support PKCS1v15 and PSS
```

Comment:
Do not like this but for supporting of two paddings

Reply:
As this is our own transaction format we can define padding we would like to use. Let's stick to PSS.

Reply:
Let's leave it as it is for now, as far as there is a lmitation now on client side. 
So later we will recearch that, and do appropriate modifications on both sides.

Reply:
*UPD* @DeV1doR 
After a brief discussion with @apadenko we decided to do the following, Add an enum to the corresponding protobuf by which we can specify the padding used here and just add a switch. This will save us resources on checking signatures which cannot be valid at all.